commands/outdated.js

```diff
@@ -28,38 +103,44 @@ const outdated /*: Outdated */ = async ({root, logger = console.log}) => {
       }
     }
   }
-  // report local discrepancies
+
+  // handle local discrepancies
   for (const name in map) {
-    const local = locals.find(local => local.meta.name === name);
+    const local = getLocal(name);
     if (local) {
       const {version} = local.meta;
-      for (const range of map[name]) {
-        if (version !== range) logger(name, range, version);
+      const outOfDate = [...map[name]].filter(
+        (consumed /*: string */) => consumed !== version
+      );
+      if (outOfDate.length > 0) {
+        results.push({name, range: outOfDate, latest: version});
       }
     }
   }
-  // then report registry discrepancies
-  for (const name in map) {
-    const local = locals.find(local => local.meta.name === name);
-    if (!local) {
-      const query = `${node} ${yarn} npm info ${name} -f version --json 2>/dev/null`;
-      let latest;
-      try {
-        const meta = JSON.parse(await exec(query));
-        latest = meta.version;
-      } catch (e) {
-        continue;
-      }
-      if (latest) {
-        for (const range of map[name]) {
-          if (!validRange(range) || !validRange(latest)) {
-            continue;
-          }
-          if (gt(latest, minVersion(range))) logger(name, range, latest);
+
+  // handle registry discrepancies
+  const info = await fetchInfo(
+    Object.keys(map).filter((pckg /*: string */) => !getLocal(pckg))
+  );
+
+  for (const name in info) {
+    const latest = info[name].version;
+    if (latest && typeof latest === 'string') {
+      for (const range of map[name]) {
+        if (!validRange(range) || !validRange(latest)) {
+          continue;
+        }
+        if (gt(latest, minVersion(range))) {
+          results.push({name, range: [range], latest});
         }
       }
     }
   }
+
+  // report discrepancies
+  for (const {name, range, latest} of results) {
```

Comment:
Minor refactor here to log at the end of processing instead of in real time.  This will make it easier to support additional formats (like a `--json` flag, for example).